### PR TITLE
edge: use golangci-lint for edge_lint

### PR DESCRIPTION
**Pull request by ziyi-yan**
For the consistency with other commands, such as cloud_lint and keadm_lint,
this PR changes Makefile command `edge_verify` to `edge_lint` and call `golangci-lint`
directly. So, edge/hack/verify.sh is no longer being used and deleted.

Resolves: #978

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature

/kind test

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: edge/Makefile [hunk 1]
```diff
@@ -16,10 +16,11 @@ export GOARCHAIUS_CONFIG_PATH=$(CURDIR)
...
 	find ${TEST_DIR} -name "*_test.go"| xargs -I{} dirname {} | uniq | xargs -I{} go test ${T} {}
 
-# verify
-.PHONY: verify
-verify:
-	bash -x hack/verify.sh
+# lint
+.PHONY: lint
+lint:
+	golangci-lint run --disable-all -E gofmt -E golint -E misspell --skip-dirs test --tests=false --deadline '10m' ./...
```
Comment (fisherxu) [inline edge/Makefile]: better to check the test as well here :)

**Comment by fisherxu** [comment]
Seems have some golint error need to fixed after change to golangci-lint

**Comment by ziyi-yan** [comment]
Finally got all lint errors fixed. @fisherxu @subpathdev PTAL

File: edge/test/integration/utils/helpers/helpers.go [hunk 2]
```diff
@@ -319,11 +312,6 @@ func HandleAddAndDeletePods(operation string, edgedpoint string, UID string, con
 		}
 		req, err = http.NewRequest(http.MethodDelete, edgedpoint, bytes.NewBuffer(respbytes))
 	}
-	if err != nil {
```
Comment (fisherxu) [inline edge/test/integration/utils/helpers/helpers.go]: how about move this err into switch,

Comment (ziyi-yan) [reply edge/test/integration/utils/helpers/helpers.go]: I read it again and found that it's actually a bug. The returned err from http.NewRequest is not visible outside the switch statement. There are also some duplication in code. I will refactor it.

Comment (fisherxu) [reply edge/test/integration/utils/helpers/helpers.go]: yes, can write a func here, also same with line 273

File: edge/test/integration/utils/helpers/helpers.go [hunk 3]
```diff
@@ -270,11 +269,6 @@ func HandleAddAndDeleteDevice(operation, testMgrEndPoint string, device dttype.D
 		}
 		req, err = http.NewRequest(http.MethodDelete, testMgrEndPoint, bytes.NewBuffer(respbytes))
 	}
-	if err != nil {
```
Comment (fisherxu) [inline edge/test/integration/utils/helpers/helpers.go]: same here

File: edge/test/integration/utils/helpers/helpers.go (hunk 2 above)
Comment (ziyi-yan) [reply edge/test/integration/utils/helpers/helpers.go]: Done with [a new separate commit](Done with a new separate commit https://github.com/kubeedge/kubeedge/pull/1009/commits/6c685fcbd855c240131ae35e89e1b0c464edbb08). PTAL

**Comment by sids-b** [comment]
@kevin-wangzefeng @fisherxu @rohitsardesai83 , golangci-lint uses GPL license. I hope it is not an issue ?

File: edge/test/integration/utils/helpers/helpers.go (hunk 2 above)
Comment (fisherxu) [reply edge/test/integration/utils/helpers/helpers.go]: Great fix!

**Comment by fisherxu** [comment]
> @kevin-wangzefeng @fisherxu @rohitsardesai83 , golangci-lint uses GPL license. I hope it is not an issue ?

Looks fine, we can use it.

**Pull request merged**